app/channel.go

```diff
@@ -596,10 +596,90 @@ func (a *App) postChannelPrivacyMessage(user *model.User, channel *model.Channel
 	return nil
 }
 
-func (a *App) RestoreChannel(channel *model.Channel) (*model.Channel, *model.AppError) {
-	if err := a.Srv.Store.Channel().Restore(channel.Id, model.GetMillis()); err != nil {
+func (a *App) RestoreChannel(channel *model.Channel, userId string) (*model.Channel, *model.AppError) {
+	ihc := make(chan store.StoreResult, 1)
+	ohc := make(chan store.StoreResult, 1)
+
+	go func() {
+		webhooks, err := a.Srv.Store.Webhook().GetIncomingByChannel(channel.Id)
+		ihc <- store.StoreResult{Data: webhooks, Err: err}
+		close(ihc)
+	}()
+
+	go func() {
+		outgoingHooks, err := a.Srv.Store.Webhook().GetOutgoingByChannel(channel.Id, -1, -1)
+		ohc <- store.StoreResult{Data: outgoingHooks, Err: err}
+		close(ohc)
+	}()
+
+	var user *model.User
+	if userId != "" {
+		var err *model.AppError
+		user, err = a.Srv.Store.User().Get(userId)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	ihcresult := <-ihc
+	if ihcresult.Err != nil {
+		return nil, ihcresult.Err
+	}
+
+	ohcresult := <-ohc
+	if ohcresult.Err != nil {
+		return nil, ohcresult.Err
+	}
+
+	incomingHooks := ihcresult.Data.([]*model.IncomingWebhook)
+	outgoingHooks := ohcresult.Data.([]*model.OutgoingWebhook)
+
+	if channel.DeleteAt == 0 {
+		err := model.NewAppError("undeleteChannel", "api.channel.undelete_channel.undeleted.app_error", nil, "", http.StatusBadRequest)
 		return nil, err
 	}
+
+	if user != nil {
+		T := utils.GetUserTranslations(user.Locale)
+
+		post := &model.Post{
+			ChannelId: channel.Id,
+			Message:   fmt.Sprintf(T("api.channel.undelete_channel.unarchived"), user.Username),
+			Type:      model.POST_CHANNEL_UNDELETED,
+			UserId:    userId,
+			Props: model.StringInterface{
+				"username": user.Username,
+			},
+		}
+
+		if _, err := a.CreatePost(post, channel, false); err != nil {
+			mlog.Error("Failed to post unarchive message", mlog.Err(err))
+		}
+	}
+
+	now := model.GetMillis()
+	for _, hook := range incomingHooks {
+		if err := a.Srv.Store.Webhook().DeleteIncoming(hook.Id, now); err != nil {
+			mlog.Error("Encountered error deleting incoming webhook", mlog.String("hook_id", hook.Id), mlog.Err(err))
+		}
+		a.InvalidateCacheForWebhook(hook.Id)
+	}
+	for _, hook := range outgoingHooks {
+		if err := a.Srv.Store.Webhook().DeleteOutgoing(hook.Id, now); err != nil {
```

Comment:
Can you elaborate why we need to delete the webhooks again? There are already deleted when the channel is archived.

Reply:
I am still learning about webhooks. 😅. you are right. We do not need to delete them.  Since we are unarchiving them. do we need to create incoming and outgoing webhooks for the channel?

Reply:
Good question. @wiersgallak should existing webhooks be restored?

Reply:
@jasonblais or @aaronrothschild Can one of you weigh in on this question, since it's more in your area of ownership?

Reply:
For the purposes of this PR, I would keep the behaviour consistent when you restore a channel [via the API](https://api.mattermost.com/#tag/channels%2Fpaths%2F~1channels~1%7Bchannel_id%7D~1restore%2Fpost) or [via the CLI](https://docs.mattermost.com/administration/command-line-tools.html#mattermost-channel-restore).

So for whatever we do there, we should do the same here, and document it.

Reply:
the API and CLI documentation doesnt say anything about webhooks.

Reply:
@allenlai18 Sorry for the late reply. You're correct, the documentation does not. Would you be open to trying what happens when you archive and restore a channel containing a webhook via the API (or CLI, whichever is easier for you)? We can then use the same approach here and document it.

Reply:
@jasonblais @hanzei @saturninoabril  when you archive a channel. a message comes up on channel for "channel is archived" and the webhooks are deleted.

when you unarchive a channel. for web the a message 'channel is unarchive' does not show up in realtime but you need to refresh. but for mobile it works. I am not too sure why, i think it has something to do with webhooks. I was hoping to get some help with that.

Reply:
>  i think it has something to do with webhooks
It's with the websocket event/message.  You may revisit how it's handled on webapp.